File: 2663.py
A = [1,0] # Base cases, starting at n=1
C = [0,1]
D = [0,3]

for n in range(3,30):
    n = n-1 # Offset
    A.append(D[n-1]+C[n-1])
    C.append(A[n-1])
    D.append(D[n-2]+2*C[n])

usernum = int(input())
while usernum != -1:
    print('    ', D[usernum-1])
    usernum = int(input())

File: POJ/2663.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n;
    int j; // Tracks offset
    vector<int> A = {1,0};
    vector<int> C = {0,1};
    vector<int> D = {0,3};

    for (int i=3; i<=30; ++i) { // Generate in range of allowed input
        j = i-1; // Offset
        A.push_back(D[j-1]+C[j-1]);
        C.push_back(A[j-1]);
        D.push_back(D[j-2]+2*C[j]);
    }

    cin >> n;
    while (n != -1) {
        cout << D[n-1] << endl; // Offset
        cin >> n;
    }

    return 0;
}
